main/src/ca/uwaterloo/flix/language/phase/Inliner.scala

```diff
@@ -148,8 +148,20 @@ object Inliner {
         else
           LiftedAst.Expression.Is(sym, e, purity, loc)
 
-      case AtomicOp.Tag(sym) => ???
-      case AtomicOp.Untag(sym) => ???
+      case AtomicOp.Tag(sym) =>
+        val List(exp) = exps
```

Comment:
This match looks fishy. I know it's _right_, but it looks fishy :)

Do you think it would be nice to have some kind of `unpackAtomicOp(1|2|3)` function that explicitly throws an ICE when there is the wrong number of exps? Up to you.

Reply:
But why do we even need to match on tag?

Reply:
We do this in GenExpression I believe so this is consistent with other uses

Reply:
If we must deal with some subset of ApplyAtomic then I think we have to match on those and use a wildcard for the rest.